### PR TITLE
Fix support for external shared Slack channels

### PR DESCRIPTION
**Pull request by reichert621**
### Description

- [x] Fixes support for external shared Slack channel webhook event payload
- [x] Fixes issue with marking conversation "read" from Slack message
- [x] Makes it possible to spawn conversation thread from primary Slack support channel by tagging it with an 👀 emoji 

### Issue

Fixes some edge cases related to external shared Slack channels

## Checklist

- [x] Everything passes when running `mix test`
- [x] Ran `mix format`
- [x] No frontend compilation warnings

### CODE REVIEW

File: lib/chat_api_web/controllers/gmail_controller.ex [hunk 1]
```diff
@@ -105,7 +105,7 @@ defmodule ChatApiWeb.GmailController do
     json(conn, %{data: %{url: url}})
   end
 
-  @spec notify_slack(Conn.t()) :: Conn.t()
+  @spec notify_slack(Plug.Conn.t()) :: Plug.Conn.t()
```
Comment (reichert621) [inline lib/chat_api_web/controllers/gmail_controller.ex:108]: (fixes dialyzer warning)

File: lib/chat_api_web/controllers/registration_controller.ex [hunk 2]
```diff
@@ -74,7 +74,7 @@ defmodule ChatApiWeb.RegistrationController do
     end
   end
 
-  @spec create(Plug.Conn.t(), map()) :: Plug.Conn.t()
+  @spec create(Conn.t(), map()) :: Conn.t()
```
Comment (reichert621) [inline lib/chat_api_web/controllers/registration_controller.ex:77]: (fixes dialyzer warning)
